planner/core/logical_plan_builder.go

```diff
@@ -1234,6 +1292,8 @@ func (g *gbyResolver) Leave(inNode ast.Node) (ast.Node, bool) {
 				ret.Accept(extractor)
 				if len(extractor.AggFuncs) != 0 {
 					err = ErrIllegalReference.GenWithStackByArgs(v.Name.OrigColName(), "reference to group function")
+				} else if ast.HasWindowFlag(ret) {
+					err = ErrIllegalReference.GenWithStackByArgs(v.Name.OrigColName(), "reference to window function")
```

Comment:
it's better to add a test for this error.

Reply:
Added in https://github.com/pingcap/tidb/pull/8630/files#diff-9bd3d43d440bb72c7bbc5c509505a553R1967